pymakr.js

```diff
@@ -1,145 +1,118 @@
         } else {
+            const PanelView = require('./lib/main/panel-view');
+            const Pymakr = require('./lib/pymakr');
+            const Pyboard = require('./lib/board/pyboard');
 
 
-            PanelView = require('./lib/main/panel-view');
-            Pymakr = require('./lib/pymakr');
-            Pyboard = require('./lib/board/pyboard');
-
-
-            pb = new Pyboard(sw)
-            v = new PanelView(pb, sw)
-            pymakr = new Pymakr({}, pb, v, sw)
-
+            const pyboard = new Pyboard(settingsWrapper)
+            const panelView = new PanelView(pyboard, settingsWrapper)
+            const pymakr = new Pymakr({}, pyboard, panelView, settingsWrapper)
+            const { terminal } = panelView
 
-            var terminal = v.terminal
-
-            var disposable = vscode.commands.registerCommand('pymakr.help', function () {
-                terminal.show()
-                pymakr.writeHelpText()
-            })
-            context.subscriptions.push(disposable);
-
-            var disposable = vscode.commands.registerCommand('pymakr.listCommands', function () {
-                v.showQuickPick()
-            })
-            context.subscriptions.push(disposable);
-
-            var disposable = vscode.commands.registerCommand('pymakr.connect', function () {
-                terminal.show()
-                pymakr.connect()
-            })
-
-            var disposable = vscode.commands.registerCommand('pymakr.run', function () {
-                terminal.show()
-                pymakr.run()
-            })
-            context.subscriptions.push(disposable);
-
-            var disposable = vscode.commands.registerCommand('pymakr.runselection', function () {
-                terminal.show()
-                pymakr.runselection()
-            })
-            context.subscriptions.push(disposable);
-
-            var disposable = vscode.commands.registerCommand('pymakr.upload', function () {
-                terminal.show()
-                pymakr.upload()
-            })
-            context.subscriptions.push(disposable);
-
-            var disposable = vscode.commands.registerCommand('pymakr.uploadFile', function () {
-                terminal.show()
-                pymakr.uploadFile()
-            })
-            context.subscriptions.push(disposable);
-
-            var disposable = vscode.commands.registerCommand('pymakr.download', function () {
-                terminal.show()
-                pymakr.download()
-            })
-            context.subscriptions.push(disposable);
-
-            var disposable = vscode.commands.registerCommand('pymakr.globalSettings', function () {
-                pymakr.openGlobalSettings()
-            })
-            context.subscriptions.push(disposable);
+            destroyHandles.push(() => panelView.destroy())
```

Comment:
this will return a destroy function, that we store to use later or fire the destroy process?

Reply:
This will add a self-destruct function for `panelView` to `destroyHandles`.

If the extension is deactivated, every function in the destroyHandles array will be called.


```javascript
module.exports = {
    activate,
    // call each function in the destroyHandles array
    deactivate: () => destroyHandles.map(destroy => destroy())
}
````

I think it might be better to rename `destroyHandles` to `onDeactivate`. What do you think?